codebase/codebase.go

```diff
@@ -0,0 +1,68 @@
+package codebase
+
+import "errors"
+
+// Codebase defines all parameters those are useful to associate Che Editor's window to a WI
+type Codebase struct {
+	Repository string `json:"repository"`
+	Branch     string `json:"branch"`
+	FileName   string `json:"filename"`
+	LineNumber int    `json:"linenumber"`
+}
```

Comment:
Yes @sbose78 This list may grow/shrink as per need.
Even `line number` is not required as of now.
I will confirm the same with Ilya once. Thanks.